crates/matrix-sdk-base/src/store/traits.rs

```diff
@@ -361,7 +361,7 @@ pub trait StateStore: AsyncTraitDeps {
         transaction_id: OwnedTransactionId,
         request: QueuedRequestKind,
         priority: usize,
-    ) -> Result<(), Self::Error>;
+    ) -> Result<MilliSecondsSinceUnixEpoch, Self::Error>;
```

Comment:
Instead of having the store return the time of creation: could it be fed by the caller, as a new function parameter? Then we can keep on returning `()` here, and we should have a bit less repetitive code in the stores.